File: bindings/lua/Polycode/LightInfo.lua
class "LightInfo"



function LightInfo:__getvar(name)
	if name == "importance" then
		return Polycore.LightInfo_get_importance(self.__ptr)
	elseif name == "position" then
		local retVal = Polycore.LightInfo_get_position(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "direction" then
		local retVal = Polycore.LightInfo_get_direction(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "type" then
		return Polycore.LightInfo_get_type(self.__ptr)
	elseif name == "diffuseColor" then
		local retVal = Polycore.LightInfo_get_diffuseColor(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Color"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "specularColor" then
		local retVal = Polycore.LightInfo_get_specularColor(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Color"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "constantAttenuation" then
		return Polycore.LightInfo_get_constantAttenuation(self.__ptr)
	elseif name == "linearAttenuation" then
		return Polycore.LightInfo_get_linearAttenuation(self.__ptr)
	elseif name == "quadraticAttenuation" then
		return Polycore.LightInfo_get_quadraticAttenuation(self.__ptr)
	elseif name == "intensity" then
		return Polycore.LightInfo_get_intensity(self.__ptr)
	elseif name == "spotlightCutoff" then
		return Polycore.LightInfo_get_spotlightCutoff(self.__ptr)
	elseif name == "spotlightExponent" then
		return Polycore.LightInfo_get_spotlightExponent(self.__ptr)
	elseif name == "shadowsEnabled" then
		return Polycore.LightInfo_get_shadowsEnabled(self.__ptr)
	elseif name == "shadowMapTexture" then
		local retVal = Polycore.LightInfo_get_shadowMapTexture(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Texture"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "lightViewMatrix" then
		local retVal = Polycore.LightInfo_get_lightViewMatrix(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Matrix4"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	end
end


function LightInfo:__setvar(name,value)
	if name == "importance" then
		Polycore.LightInfo_set_importance(self.__ptr, value)
		return true
	elseif name == "position" then
		Polycore.LightInfo_set_position(self.__ptr, value.__ptr)
		return true
	elseif name == "direction" then
		Polycore.LightInfo_set_direction(self.__ptr, value.__ptr)
		return true
	elseif name == "type" then
		Polycore.LightInfo_set_type(self.__ptr, value)
		return true
	elseif name == "diffuseColor" then
		Polycore.LightInfo_set_diffuseColor(self.__ptr, value.__ptr)
		return true
	elseif name == "specularColor" then
		Polycore.LightInfo_set_specularColor(self.__ptr, value.__ptr)
		return true
	elseif name == "constantAttenuation" then
		Polycore.LightInfo_set_constantAttenuation(self.__ptr, value)
		return true
	elseif name == "linearAttenuation" then
		Polycore.LightInfo_set_linearAttenuation(self.__ptr, value)
		return true
	elseif name == "quadraticAttenuation" then
		Polycore.LightInfo_set_quadraticAttenuation(self.__ptr, value)
		return true
	elseif name == "intensity" then
		Polycore.LightInfo_set_intensity(self.__ptr, value)
		return true
	elseif name == "spotlightCutoff" then
		Polycore.LightInfo_set_spotlightCutoff(self.__ptr, value)
		return true
	elseif name == "spotlightExponent" then
		Polycore.LightInfo_set_spotlightExponent(self.__ptr, value)
		return true
	elseif name == "shadowsEnabled" then
		Polycore.LightInfo_set_shadowsEnabled(self.__ptr, value)
		return true
	elseif name == "lightViewMatrix" then
		Polycore.LightInfo_set_lightViewMatrix(self.__ptr, value.__ptr)
		return true
	end
	return false
end


function LightInfo:__delete()
	if self then Polycore.delete_LightInfo(self.__ptr) end
end

File: bindings/javascript/Polycode/LightInfo.js
function LightInfo() {
	Object.defineProperties(this, {
		'importance': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_importance, set: LightInfo.prototype.__set_importance},
		'position': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_position, set: LightInfo.prototype.__set_position},
		'direction': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_direction, set: LightInfo.prototype.__set_direction},
		'type': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_type, set: LightInfo.prototype.__set_type},
		'diffuseColor': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_diffuseColor, set: LightInfo.prototype.__set_diffuseColor},
		'specularColor': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_specularColor, set: LightInfo.prototype.__set_specularColor},
		'constantAttenuation': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_constantAttenuation, set: LightInfo.prototype.__set_constantAttenuation},
		'linearAttenuation': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_linearAttenuation, set: LightInfo.prototype.__set_linearAttenuation},
		'quadraticAttenuation': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_quadraticAttenuation, set: LightInfo.prototype.__set_quadraticAttenuation},
		'intensity': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_intensity, set: LightInfo.prototype.__set_intensity},
		'spotlightCutoff': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_spotlightCutoff, set: LightInfo.prototype.__set_spotlightCutoff},
		'spotlightExponent': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_spotlightExponent, set: LightInfo.prototype.__set_spotlightExponent},
		'shadowsEnabled': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_shadowsEnabled, set: LightInfo.prototype.__set_shadowsEnabled},
		'shadowMapTexture': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_shadowMapTexture, set: LightInfo.prototype.__set_shadowMapTexture},
		'lightViewMatrix': { enumerable: true, configurable: true, get: LightInfo.prototype.__get_lightViewMatrix, set: LightInfo.prototype.__set_lightViewMatrix}
	})
}
LightInfo.prototype.__get_importance = function() {
	return Polycode.LightInfo__get_importance(this.__ptr)
}

LightInfo.prototype.__set_importance = function(val) {
	Polycode.LightInfo__set_importance(this.__ptr, val)
}

LightInfo.prototype.__get_position = function() {
	var retVal = new Vector3()
	retVal.__ptr = 	Polycode.LightInfo__get_position(this.__ptr)
	return retVal
}

LightInfo.prototype.__set_position = function(val) {
	Polycode.LightInfo__set_position(this.__ptr, val.__ptr)
}

LightInfo.prototype.__get_direction = function() {
	var retVal = new Vector3()
	retVal.__ptr = 	Polycode.LightInfo__get_direction(this.__ptr)
	return retVal
}

LightInfo.prototype.__set_direction = function(val) {
	Polycode.LightInfo__set_direction(this.__ptr, val.__ptr)
}

LightInfo.prototype.__get_type = function() {
	return Polycode.LightInfo__get_type(this.__ptr)
}

LightInfo.prototype.__set_type = function(val) {
	Polycode.LightInfo__set_type(this.__ptr, val)
}

LightInfo.prototype.__get_diffuseColor = function() {
	var retVal = new Color()
	retVal.__ptr = 	Polycode.LightInfo__get_diffuseColor(this.__ptr)
	return retVal
}

LightInfo.prototype.__set_diffuseColor = function(val) {
	Polycode.LightInfo__set_diffuseColor(this.__ptr, val.__ptr)
}

LightInfo.prototype.__get_specularColor = function() {
	var retVal = new Color()
	retVal.__ptr = 	Polycode.LightInfo__get_specularColor(this.__ptr)
	return retVal
}

LightInfo.prototype.__set_specularColor = function(val) {
	Polycode.LightInfo__set_specularColor(this.__ptr, val.__ptr)
}

LightInfo.prototype.__get_constantAttenuation = function() {
	return Polycode.LightInfo__get_constantAttenuation(this.__ptr)
}

LightInfo.prototype.__set_constantAttenuation = function(val) {
	Polycode.LightInfo__set_constantAttenuation(this.__ptr, val)
}

LightInfo.prototype.__get_linearAttenuation = function() {
	return Polycode.LightInfo__get_linearAttenuation(this.__ptr)
}

LightInfo.prototype.__set_linearAttenuation = function(val) {
	Polycode.LightInfo__set_linearAttenuation(this.__ptr, val)
}

LightInfo.prototype.__get_quadraticAttenuation = function() {
	return Polycode.LightInfo__get_quadraticAttenuation(this.__ptr)
}

LightInfo.prototype.__set_quadraticAttenuation = function(val) {
	Polycode.LightInfo__set_quadraticAttenuation(this.__ptr, val)
}

LightInfo.prototype.__get_intensity = function() {
	return Polycode.LightInfo__get_intensity(this.__ptr)
}

LightInfo.prototype.__set_intensity = function(val) {
	Polycode.LightInfo__set_intensity(this.__ptr, val)
}

LightInfo.prototype.__get_spotlightCutoff = function() {
	return Polycode.LightInfo__get_spotlightCutoff(this.__ptr)
}

LightInfo.prototype.__set_spotlightCutoff = function(val) {
	Polycode.LightInfo__set_spotlightCutoff(this.__ptr, val)
}

LightInfo.prototype.__get_spotlightExponent = function() {
	return Polycode.LightInfo__get_spotlightExponent(this.__ptr)
}

LightInfo.prototype.__set_spotlightExponent = function(val) {
	Polycode.LightInfo__set_spotlightExponent(this.__ptr, val)
}

LightInfo.prototype.__get_shadowsEnabled = function() {
	return Polycode.LightInfo__get_shadowsEnabled(this.__ptr)
}

LightInfo.prototype.__set_shadowsEnabled = function(val) {
	Polycode.LightInfo__set_shadowsEnabled(this.__ptr, val)
}

LightInfo.prototype.__get_shadowMapTexture = function() {
	var retVal = new Texture()
	retVal.__ptr = 	Polycode.LightInfo__get_shadowMapTexture(this.__ptr)
	return retVal
}

LightInfo.prototype.__set_shadowMapTexture = function(val) {
	Polycode.LightInfo__set_shadowMapTexture(this.__ptr, val.__ptr)
}

LightInfo.prototype.__get_lightViewMatrix = function() {
	var retVal = new Matrix4()
	retVal.__ptr = 	Polycode.LightInfo__get_lightViewMatrix(this.__ptr)
	return retVal
}

LightInfo.prototype.__set_lightViewMatrix = function(val) {
	Polycode.LightInfo__set_lightViewMatrix(this.__ptr, val.__ptr)
}

Duktape.fin(LightInfo.prototype, function (x) {
	if (x === LightInfo.prototype) {
		return;
	}
	Polycode.LightInfo__delete(x.__ptr)
})
